cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -0,0 +1,192 @@
+""" Code to allow module store to interface with courseware index """
+from __future__ import absolute_import
+
+from datetime import timedelta
+import logging
+
+from django.conf import settings
+from django.utils.translation import ugettext as _
+from eventtracking import tracker
+from xmodule.modulestore import ModuleStoreEnum
+from search.search_engine_base import SearchEngine
+
+
+# Use default index and document names for now
+INDEX_NAME = "courseware_index"
+DOCUMENT_TYPE = "courseware_content"
+REINDEX_AGE = timedelta(0, 60)
+
+log = logging.getLogger('edx.modulestore')
+
+
+def indexing_is_enabled():
+    """
+    Checks to see if the indexing feature is enabled
+    """
+    return settings.FEATURES.get('ENABLE_COURSEWARE_INDEX', False)
+
+
+class SearchIndexingError(Exception):
+    """ Indicates some error(s) occured during indexing """
+
+    def __init__(self, message, error_list):
+        super(SearchIndexingError, self).__init__(message)
+        self.error_list = error_list
+
+
+class CoursewareSearchIndexer(object):
+    """
+    Class to perform indexing for courseware search from different modulestores
+    """
+
+    @staticmethod
+    def index_course(modulestore, course_key, triggered_at=None, reindex_age=REINDEX_AGE):
```

Comment:
@martynjames nit: this function is big. It's not actually big per se, but it contains definitions of two helper functions, and those are big. As a result it's a bit hard to read it. Maybe it's worth promoting those helpers to actual "private" functions?